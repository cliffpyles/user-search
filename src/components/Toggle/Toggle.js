import React, { useState } from 'react'
import './Toggle.css'

export default function Toggle({ toggledOn = false, onClick = () => {} }) {
  const [toggled, setToggle] = useState(toggledOn)
  const handleChange = () => {
    setToggle(!toggled)
  }
  return (
    <label className="Toggle" onClick={onClick}>
      <input type="checkbox" checked={toggled} onChange={handleChange} />
    </label>
  )
}
