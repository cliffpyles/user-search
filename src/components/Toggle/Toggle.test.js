import React from 'react'
import Toggle from './'

describe('Toggle', () => {
  test('can be rendered', () => {
    const component = render(<Toggle />)

    expect(component).toMatchSnapshot()
  })

  test('can be rendered with toggle on', () => {
    const component = render(<Toggle toggledOn />)

    expect(component).toMatchSnapshot()
  })

  test('can be toggled', () => {
    const handleClick = jest.fn()
    const component = render(<Toggle onClick={handleClick} />)

    component.click()

    expect(handleClick).toHaveBeenCalled()
  })
})
