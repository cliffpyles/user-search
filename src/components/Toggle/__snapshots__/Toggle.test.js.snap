// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle can be rendered 1`] = `
<div>
  <label
    class="Toggle"
  >
    <input
      type="checkbox"
    />
  </label>
</div>
`;

exports[`Toggle can be rendered with toggle on 1`] = `
<div>
  <label
    class="Toggle"
  >
    <input
      checked=""
      type="checkbox"
    />
  </label>
</div>
`;
